src/indicator.js

```diff
@@ -29,167 +29,203 @@ const St = imports.gi.St;
+const Gio = imports.gi.Gio;
 const Util = imports.misc.util;
 const Mainloop = imports.mainloop;
 const Shell = imports.gi.Shell;
 const UPower = imports.gi.UPowerGlib;
-const Gio = imports.gi.Gio;
-const GObject = imports.gi.GObject;
 const Config = imports.misc.config;
 
-const Gettext = imports.gettext.domain('gnome-shell-extension-cpupower');
+const Gettext = imports.gettext.domain("gnome-shell-extension-cpupower");
 const _ = Gettext.gettext;
 
 const ExtensionUtils = imports.misc.extensionUtils;
 const Me = ExtensionUtils.getCurrentExtension();
+const EXTENSIONDIR = Me.dir.get_path();
+const utils = Me.imports.src.utils;
+const Cpufreqctl = Me.imports.src.utils.Cpufreqctl;
 
 const Slider = Me.imports.src.slider2;
 const CPUFreqProfile = Me.imports.src.profile.CPUFreqProfile;
 const baseindicator = Me.imports.src.baseindicator;
 const CPUFreqProfileButton = Me.imports.src.profilebutton.CPUFreqProfileButton;
-const checkCpuLimit = Me.imports.src.utils.checkCpuLimit;
 
-const LASTSETTINGS = GLib.get_user_cache_dir() + '/cpupower.last-settings';
-const PKEXEC = GLib.find_program_in_path('pkexec');
-const CONFIG = Me.imports.src.config;
+const LASTSETTINGS = `${GLib.get_user_cache_dir()}/cpupower.last-settings`;
 
+/* exported CPUFreqIndicator */
 var CPUFreqIndicator = class CPUFreqIndicator extends baseindicator.CPUFreqBaseIndicator {
-    constructor() {
+    constructor(onConstructed) {
         super();
-        this.cpuMinLimit = checkCpuLimit("min");
-        this.cpuMaxLimit = checkCpuLimit("max");
         this.cpufreq = 800;
         this.cpucount = 0;
         this.isTurboBoostActive = true;
         this.isAutoSwitchActive = true;
-        this.minVal = this.cpuMinLimit;
-        this.maxVal = this.cpuMinLimit;
+        this.minVal = 0;
+        this.maxVal = 100;
 
         // read the cached settings file.
-        if(GLib.file_test(LASTSETTINGS, GLib.FileTest.EXISTS))
-        {
-            let lines = Shell.get_file_contents_utf8_sync(LASTSETTINGS).split('\n');
-            if(lines.length > 3)
-            {
+        if (GLib.file_test(LASTSETTINGS, GLib.FileTest.EXISTS)) {
+            let lines = Shell.get_file_contents_utf8_sync(LASTSETTINGS).split("\n");
+            if (lines.length > 3) {
                 this.minVal = parseInt(lines[0]);
                 this.maxVal = parseInt(lines[1]);
-                this.isTurboBoostActive = (lines[2].indexOf('true') > -1);
-                this.isAutoSwitchActive = (lines[3].indexOf('true') > -1);
+                this.isTurboBoostActive = lines[2].indexOf("true") > -1;
+                this.isAutoSwitchActive = lines[3].indexOf("true") > -1;
 
-                this._updateMin();
-                this._updateMax();
-                this._updateTurbo();
-                this._updateAutoSwitch();
+                this.updateMin();
+                this.updateMax();
+                this.updateTurbo();
+                this.updateAutoSwitch();
             }
         } else {
```

Comment:
This is not working anymore, most likely it's again another race condition...

What happens:
After login the minimum and maximum frequency are not restored but are always `min: hardware minimum` and `max: 100`, whereas the turbo boost setting is applied from the last session.

What is expected:
Restoring minimum and maximum frequency of last boot just works.